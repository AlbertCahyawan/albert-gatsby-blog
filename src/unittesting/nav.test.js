import * as React from "react"
import { fireEvent, render } from "@testing-library/react" 
import renderer from "react-test-renderer"
import Nav from '../components/core/nav' 
const blogData =[
  { 
    title: 'Hello, World!',
    date: '2 Feb 2021',
    content: `<div>hello Content</div>` 
  }
]
describe(`Navigation`, () => {
  it(`renders siteTitle link`, () => { 
    const { getByText } = render( <Nav bdata={blogData} /> )
    const title = getByText(blogData[0].title)
    expect(title).toBeInTheDocument()
  })

   const Title = () => <Nav />
   test("Displays the correct title", () => {
     const { getByTestId } = render(<Title />)
     // Assertion 
     expect(getByTestId("blog-title")).toHaveTextContent("Blog")
     // --> Test will pass
   }) 

   it("renders correctly", () => { 
    const tree = renderer.create(<Nav bdata={blogData} />).toJSON()
    expect(tree).toMatchSnapshot() 

  })
})


 
