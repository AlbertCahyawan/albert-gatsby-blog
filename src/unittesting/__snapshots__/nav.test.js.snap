// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<ul
  className="sc-bdnxRM gAuIVV"
>
  <li
    className="sc-gtsrHT bEQcMT"
    selected={false}
  >
     
    <a
      href="https://albertcahyawan.com/"
    >
      Home
    </a>
     
  </li>
  <li
    className="sc-gtsrHT eMcDLI"
    selected={true}
  >
    <a
      aria-hidden="true"
      data-testid="blog-title"
      href="/"
      onClick={[Function]}
    >
      Blog
    </a>
  </li>
  <div
    className="sc-dlnjwi fGwqfx"
  >
    <ul
      className="sc-hKFxyN laOluZ"
    >
      <li>
        <a
          href="/articles/0"
        >
           
          Hello, World!
        </a>
      </li>
    </ul>
  </div>
</ul>
`;
